### PR TITLE
Port: remove useless appendVertices below innner/left join

### PR DESCRIPTION
**Pull request by jievince**
## What type of PR is this?
- [ ] bug
- [ ] feature
- [x] enhancement

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

File: src/common/expression/PredicateExpression.cpp [hunk 1]
```diff
@@ -19,11 +19,14 @@ std::unordered_map<std::string, PredicateExpression::Type> PredicateExpression::
...
       << "actual kind: " << collection_->kind() << ", toString: " << toString();
 
-  if (collection_->kind() == Expression::Kind::kLabelTagProperty) {
-    result_ = !collection_->eval(ctx).isNull();
+  if (collection_->kind() == Expression::Kind::kLabelTagProperty ||
+      collection_->kind() == Expression::Kind::kTagProperty) {
+    auto v = collection_->eval(ctx);
+    result_ = (!v.isNull()) && (!v.empty());
     return result_;
   }
```
Comment (jievince) [inline src/common/expression/PredicateExpression.cpp:32]: Copied from nebula-ent.

File: tests/tck/features/optimizer/RemoveAppendVerticesBelowJoinRule.feature [hunk 2]
```diff
@@ -15,10 +15,10 @@ Feature: Optimize left join predicate
...
       WITH friendTeam, count(friend2) AS numFriends
       RETURN
-        id(friendTeam) AS teamId,
-        friendTeam.team.name AS teamName,
-        numFriends
-        ORDER BY teamName DESC
+      id(friendTeam) AS teamId,
+      friendTeam.team.name AS teamName,
+      numFriends
+      ORDER BY teamName DESC
```
Comment (yixinglu) [inline tests/tck/features/optimizer/RemoveAppendVerticesBelowJoinRule.feature:21]: only fix indent of `ORDER BY`

**Review by yixinglu — COMMENTED** [review]
Good JOB!

**Review by yixinglu — APPROVED** [review]

File: tests/tck/features/optimizer/RemoveAppendVerticesBelowJoinRule.feature (hunk 2 above)
Comment (jievince) [reply tests/tck/features/optimizer/RemoveAppendVerticesBelowJoinRule.feature:21]: My fault.

**Review by nevermore3 — APPROVED** [review]

**Review by czpmango — APPROVED** [review]

**Pull request merged**